lib/queueservice.js

```diff
@@ -0,0 +1,82 @@
+import xml2js from 'xml2js';
+import Debug from 'debug';
+import request from 'superagent-promise';
+
+let debug = Debug('taskcluster-docker-worker:queueService');
+
+export default class TaskQueue {
+  constructor(config) {
+    this.queues;
+    this.workerType = config.workerType;
+    this.provisionerId = config.provisionerId;
+    this.client = config.queue;
+    this.log = config.log;
+  }
+
+  async getQueues() {
+    if (!this.queues || (new Date(this.queues.expires)).getTime() > Date.now()) {
+      this.queues = await this.client.pollTaskUrls(this.provisionerId, this.workerType);
+    }
+    return this.queues.queues;
+  }
+
+  async getTasksFromQueue(queue, numberOfTasks) {
+    let tasks = [];
+    let uri = `${queue.signedPollUrl}&numofmessages=${numberOfTasks}`;
+    debug(`requesting: ${uri}`);
+    let response = await request
+      .get(uri)
+      .buffer()
+      .end();
+
+    let xml = await new Promise((accept, reject) => {
```

Comment:
this is not terrible but I am tempted to say use denodeify and make parseString a const instead 

```
const parseXmlString = denodeify(xml2js.parseString.bind(xml2js))
```